python/pyspark/testing/utils.py

```diff
@@ -382,18 +388,23 @@ def assert_rows_equal(rows1: List[Row], rows2: List[Row]):
         try:
             # rename duplicate columns for sorting
             renamed_df = df.toDF(*[f"_{i}" for i in range(len(df.columns))])
-            renamed_expected = expected.toDF(*[f"_{i}" for i in range(len(expected.columns))])
-
             df = renamed_df.sort(renamed_df.columns).toDF(*df.columns)
-            expected = renamed_expected.sort(renamed_expected.columns).toDF(*expected.columns)
+            if isinstance(expected, List):
+                expected.sort()
```

Comment:
```py
expected = sorted(expected)
```

to avoid updating the input.

Reply:
Btw, I'm wondering Python's `sort` or `sorted` works as same as `df.sort()`. 
We might want to ask users to make it already sorted when they specify a list here.
cc @allisonwang-db

Reply:
There is a flag  `check_row_orders`. If this flag is False, we will sort both the input data frame and the expected list of Rows.

Reply:
We can convert the input data frame to a list and sort both lists of rows.